perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2425,6 +2425,19 @@ public Member sponsorMember(PerunSession session, Member sponsoredMember, User s
 		getMembersManagerImpl().addSponsor(session, sponsoredMember, sponsor);
 		getPerunBl().getAuditer().log(session, new SponsorshipEstablished(sponsoredMember, sponsor));
```

Comment:
I would suggest moving this log to the end of the method. If any of the next operations fail, the process of the AuditEvent would be done unnecessarily.

Reply:
You are right. Done.